ollama/_client.py

```diff
@@ -617,12 +619,14 @@ def __init__(self, host: Optional[str] = None, **kwargs) -> None:
     super().__init__(httpx.AsyncClient, host, **kwargs)
 
   async def _request_raw(self, *args, **kwargs):
-    r = await self._client.request(*args, **kwargs)
     try:
+      r = await self._client.request(*args, **kwargs)
       r.raise_for_status()
+      return r
     except httpx.HTTPStatusError as e:
       raise ResponseError(e.response.text, e.response.status_code) from None
-    return r
+    except httpx.ConnectError:
+      raise ResponseError('Failed to connect to Ollama. Please check that Ollama is downloaded, running and accessible. https://ollama.com/download', 503) from None
```

Comment:
Shouldn't a 503 code be provided by the server? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503 I wouldn't introduce any codes that aren't actually served by the API

Reply:
Yeah I'm debating if we should not have this as a response error since it's not even a response that is failing it's the connection from httpx itself

Reply:
Found a `ConnectionError` throwable which I think makes more sense in this setting